optimum/exporters/neuron/convert.py

```diff
@@ -458,27 +458,33 @@ def export_neuronx(
 
 
 def add_stable_diffusion_compiler_args(config, compiler_args):
-    if hasattr(config._config, "_name_or_path"):
-        sd_components = ["text_encoder", "vae", "vae_encoder", "vae_decoder"]
-        if any(component in config._config._name_or_path.lower() for component in sd_components):
-            compiler_args.extend(["--enable-fast-loading-neuron-binaries"])
-        # unet
-        if "unet" in config._config._name_or_path.lower():
-            # SDXL unet doesn't support fast loading neuron binaries
-            if "stable-diffusion-xl" not in config._config._name_or_path.lower():
-                compiler_args.extend(["--enable-fast-loading-neuron-binaries"])
-            compiler_args.extend(["--model-type=unet-inference"])
+    # Combine the model name and its path to identify which is the subcomponent in Stable Diffusion pipeline
+    identifier = getattr(config._config, "_name_or_path", "") + " " + getattr(config._config, "_class_name", "")
+    identifier = identifier.lower()
+
+    sd_components = ["text_encoder", "vae", "vae_encoder", "vae_decoder"]
+    if any(component in identifier for component in sd_components):
+        compiler_args.append("--enable-fast-loading-neuron-binaries")
+    # unet
+    if "unet" in identifier:
+        sdxl_ids = ["stable-diffusion-xl", "sdxl"]
+        # SDXL unet doesn't support fast loading neuron binaries
+        if not any(sdxl_id in identifier for sdxl_id in sdxl_ids):
```

Comment:
is there a way we can utilize `--task` option to do this check? or maybe check the model class name too

because the model id or folder name of a SDXL model might not always contain one of these identifiers

Reply:
Hi @neo, thanks for the suggestion. It sounds good to me, I will prioritize the task in neuron config than the identifier in a coming PR (there is a lack of consideration on some edge cases, thanks for pointing it out.